source/connection-options.txt

```diff
@@ -0,0 +1,592 @@
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: connection string, URI, server, settings, configure
+
+Overview
+--------
+
+This section describes the MongoDB connection and authentication options
+available in the {+driver-short+}. You can configure your connection using either
+the connection URI or arguments to the ``MongoClient`` constructor.
+
+.. _java-rs-connection-uri:
+
+Using the Connection URI
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you pass a connection URI to the ``MongoClient`` constructor, you can include
+connection options in the string as ``<name>=<value>`` pairs. In the following example,
+the connection URI contains the ``connectTimeoutMS`` option with a value of ``60000``
+and the ``tls`` option with a value of ``true``:
+
+.. code-block:: java
+
+    public class MongoConnection {
+    public static void main(String[] args) {
+        String uri = "mongodb://<hostname>:<port>/?connectTimeoutMS=60000&tls=true";
+        try (MongoClient client = MongoClients.create(uri)) {
+                //use client here
+            }
+        }
+    }
+
+.. _java-rs-mongo-client-settings:
+
+Using a ``MongoClient``
+~~~~~~~~~~~~~~~~~~~~~~~
+
+You can pass connection options as arguments to the ``MongoClient`` constructor
+instead of including them in your connection URI.
+Configuring the connection this way makes it easier to
+change settings at runtime and helps you catch errors during compilation.
+The following example shows how to use the ``MongoClient`` constructor to set
+connection options:
+
+.. code-block:: java
+
+    public class MongoConnection {
+        public static void main(String[] args) {
+            String uri = "mongodb://<hostname>:<port>";
+            try (MongoClient client = MongoClients.create(uri, connectTimeoutMS=60000, tls=True) {
+                //use client here
+            }
+        }
+    }
+
+Connection Options
+------------------
+
+The following sections describe the connection options available in the
+{+driver-short+}.
+
+Authentication
+~~~~~~~~~~~~~~
+
+.. list-table::
+   :header-rows: 1
+   :widths: 30 70
+
+   * - Connection Option
+     - Description
+
+   * - **authMechanism**
+     - | Sets the mechanism the {+driver-short+} uses to authenticate the
+       | application. Valid options are defined in `Class ConnectionString. <{+api+}/mongodb-driver-core/com/mongodb/ConnectionString.html>`__
+       |
+       | **Data Type**: {+string-data-type+}
+       | **Default**: ``"SCRAM-SHA-256"`` when connecting to MongoDB v4.0 or later.
+       | ``"SCRAM-SHA-1"`` when connecting to MongoDB v3.0 through v3.13.
+       | **MongoClient Example**: ``authMechanism = "MONGODB-X509"``
+       | **Connection URI Example**: ``authMechanism=MONGODB-X509``
+  
+   * - **authMechanismProperties**
+     - | Allows you to set authentication mechanism properties on the connection
```

Comment:
This was the definition that the api docs gave: "This option allows authentication mechanism properties to be set on the connection string."

Reply:
https://mongodb.github.io/mongo-java-driver/5.1/apidocs/mongodb-driver-core/com/mongodb/ConnectionString.html 

i wrote it based on the api docs, but it is also possible the api docs are wrong, because you're correct that the pymongo docs say something else, and all the methods should be the same in different drivers/have the same functionality

Reply:
Oh interesting. Maybe defer to the tech review for this, but it seems like the Pymongo wording is more accurate